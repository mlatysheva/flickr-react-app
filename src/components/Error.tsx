function Error() {
  return (
    <div className="main">
      <h1>The requested page does not exist</h1>
    </div>
  );
}

export default Error;
