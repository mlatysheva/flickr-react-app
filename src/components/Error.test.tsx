import { render, screen } from '@testing-library/react';
import Error from './Error';
import { MemoryRouter } from 'react-router-dom';

test('renders text', () => {
  render(
    <MemoryRouter>
      <Error />
    </MemoryRouter>);
  const text = screen.getByText(/The requested page does not exist/i);
  expect(text).toBeInTheDocument();
});

test('renders Error', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <Error />
    </MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
});