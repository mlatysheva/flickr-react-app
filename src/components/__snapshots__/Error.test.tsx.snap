// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Error 1`] = `
<DocumentFragment>
  <div
    class="main"
  >
    <h1>
      The requested page does not exist
    </h1>
  </div>
</DocumentFragment>
`;
